### PR TITLE
Add team season statistics and admin season management

### PR DESCRIPTION
**Pull request by HighwayofLife**
## Summary
- Add team season statistics API endpoint with career and current season stats
- Implement admin UI for season management
- Enhanced team detail page with interactive statistics display

## Changes
- **New Features:**
  - `/v1/teams/{id}/stats` API endpoint for team statistics including career totals, current season, and recent games
  - Admin season management page at `/admin/seasons` for creating, editing, and activating seasons
  - Enhanced team detail page with tabbed interface showing roster, season stats, and career stats
  
- **UI Improvements:**
  - Added season selector dropdown to team detail page
  - Interactive statistics display with shooting percentages and per-game averages
  - Recent games history with win/loss indicators and detailed stats
  
- **Testing:**
  - Added comprehensive integration tests for team detail page functionality
  - Added unit tests for new team stats API endpoint
  - Test coverage for season statistics calculations and edge cases

## Test plan
- [x] Run unit tests: `make test-unit`
- [x] Run integration tests: `make test-integration`
- [x] Manual testing of team detail page statistics display
- [ ] Verify season management admin functionality
- [ ] Test API endpoint response times with large datasets

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: app/web_ui/routers/pages.py [hunk 1]
```diff
@@ -372,6 +379,12 @@ async def admin_users_page(request: Request):
     return templates.TemplateResponse("admin/users.html", {"request": request, "title": "User Management"})
 
 
+@router.get("/admin/seasons", response_class=HTMLResponse)
```
[the remark this answers is not in the document]
Comment (HighwayofLife) [reply app/web_ui/routers/pages.py:382]: doesn't need it, we already handle that elsewhere.
